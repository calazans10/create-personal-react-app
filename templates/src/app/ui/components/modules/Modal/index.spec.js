import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Modal } from './index';

describe('Modal', () => {
  describe('when is visible', () => {
    it('renders without crashing', () => {
      const props = {
        children: <p>Test</p>,
        isVisible: true,
        onHideModal: jest.fn(),
      };
      const wrapper = shallow(<Modal {...props} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
