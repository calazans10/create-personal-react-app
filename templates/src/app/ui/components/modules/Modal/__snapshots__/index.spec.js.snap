// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal when is visible renders without crashing 1`] = `
<Portal>
  <div
    className="modal modal--visible"
  >
    <div
      className="modal__dialog"
    >
      <div
        className="modal__content"
      >
        <p>
          Test
        </p>
        <button
          aria-label="Close modal"
          className="modal__trigger"
          onClick={[MockFunction]}
          type="button"
        >
          <span
            className="visually-hidden"
          >
            x
          </span>
        </button>
      </div>
    </div>
  </div>
</Portal>
`;
